content/blog/building-images-bake/index.md

```diff
@@ -0,0 +1,169 @@
+- Another Bake file, but this time a local one, this one is to overwrite the previous one.
+
+Baking time: 5 minutes.
+
+## Instructions
+
+### Step 1: Prepare local Bake file
+
+In a local file with name [bake.hcl](bake.hcl), we add the following content, which is a simple Bake file that will build a custom image
+
+```hcl
+extensions = [
+  "pgvector",
+]
+target "myimage" {
+  dockerfile-inline = <<EOT
+ARG BASEIMAGE="ghcr.io/cloudnative-pg/postgresql:16.9-standard-bookworm"
+FROM $BASEIMAGE AS myimage
+ARG EXTENSIONS
+USER root
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends $EXTENSIONS && \
+    apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false && \
+    rm -rf /var/lib/apt/lists/* /var/cache/* /var/log/*
+USER 26
+EOT
+
+  matrix = {
+    tgt = [
+      "myimage"
+    ]
+    pgVersion = [
+      "16.9",
+      "17.5",
+    ]
+  }
+  name = "postgresql-${index(split(".",cleanVersion(pgVersion)),0)}-standard-bookworm"
+  target = "${tgt}"
+  args = {
+    BASE_IMAGE = "ghcr.io/cloudnative-pg/postgresql:${cleanVersion(pgVersion)}-standard-bookworm",
+    EXTENSIONS = "${getExtensionsString(pgVersion, extensions)}",
+  }
+}
+```
+
+There are a few things that we should remark here:
+
+- The `extensions` variable is a list of extensions that we want to include in the image. In our recipe we are using `pg_vector`,
+  but you can add any other extension that you want to include in the image.
+- The `dockerfile-inline` variable contains our Dockerfile definition, which cannot be used remotely. We will explain more about this later.
+- The `target` and the `tgt` have the same name, you can use whatever you want here as a name
+- The `pgVersion` variable is a list that contains basically the MAJOR.MINOR version of PostgreSQL
+- The `name` is the name that we will use later to refer to one element of the matrix that we created
+- The variable `args` lists all the arguments passed to the Dockerfile. We will talk more about this later.
+- The function `getExtensionsString()` is inherited from the Bake file that we reference in the [Ingredients](#ingredients) section
+
+### Step 2: Build the image
+
+We can now build the image using the following command:
+
+```bash
+docker buildx bake -f docker-bake.hcl -f cwd://bake.hcl "https://github.com/cloudnative-pg/postgres-containers.git" myimage
+```
+
+This will, by default, build the image for the bake matrix we previously created, and will try to push the image to the registry at
+`localhost:5000`, which is the default registry defined for testing environments in the parent Bake file. Let's explain the full command:
+
+As is defined in the [Bake documentation about remote definitions](https://docs.docker.com/build/bake/remote-definition/)
+you can use a remote Bake definition with all the functions and default targets, and attach another local one that you can use to override
+all the default values.
```

Comment:
First of all, this is a bit redundant, so I have a suggestion.
Second: if you override ALL the default values, why even have a base file?
```suggestion
you can use a remote Bake definition with all the functions and default targets, and attach another local one to override
all the default values.
```